dev/tools/controllerbuilder/pkg/llm/options.go

```diff
@@ -41,6 +44,35 @@ func (o *Options) InitDefaults() {
 	o.Model = model
 }
 
+func (o *Options) InitDefaultsWithLatestModelIfUnset(ctx context.Context, llmClient gollm.Client) error {
+	model := os.Getenv("LLM_MODEL")
+
+	modelNameSubString := "gemini-2.5-pro"
```

Comment:
If I read the code correctly, it does not set the latest model as the default. It still requires manual code changes everytime a new model comes, right?

Reply:
Yes for now, not really for the future. We can have the regex to not only match `2.5` but also any `X.Y` and pick any that is above 2.5.